lib/active_merchant/billing/gateways/payu_latam.rb

```diff
@@ -0,0 +1,393 @@
+require 'digest/md5'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class PayuLatamGateway < Gateway
+      self.display_name = "PayU Latam"
+      self.homepage_url = "http://www.payulatam.com"
+
+      self.test_url = "https://sandbox.api.payulatam.com/payments-api/4.0/service.cgi"
+      self.live_url = "https://api.payulatam.com/payments-api/4.0/service.cgi"
+
+      self.supported_countries = ["AR", "BR", "CO", "MX", "PA", "PE"]
+      self.default_currency = "USD"
+      self.money_format = :dollars
+      self.supported_cardtypes = [:visa, :master, :american_express, :diners_club]
+
+      BRAND_MAP = {
+        "visa" => "VISA",
+        "master" => "MASTERCARD",
+        "american_express" => "AMEX",
+        "diners_club" => "DINERS"
+      }
+
+      def initialize(options={})
+        requires!(options, :merchant_id, :account_id, :api_login, :api_key)
+        super
+      end
+
+      def purchase(amount, payment_method, options={})
+        post = {}
+
+        add_credentials(post, 'SUBMIT_TRANSACTION')
```

Comment:
Looks like there's some logic that shows up in `purchase` and `authorize`.  Perhaps it can be refactored a bit to DRY things up some.